app/services/journey_logger_service.rb

```diff
@@ -5,8 +5,11 @@ def call(assessment_id, calculation_result, check, portal_user_office_code, cook
 
       attributes = build_attributes(calculation_result, check, portal_user_office_code)
       CompletedUserJourney.transaction do
-        if (journey = CompletedUserJourney.find_by(assessment_id:))
-          journey.update!(attributes)
+        early_result_type = check.early_ineligible_result?
+        existing_journey = CompletedUserJourney.find_by(assessment_id:, early_eligibility_result: early_result_type || nil)
```

Comment:
This looks really nice and compact - I would probably change the variable name of early_result_type as it is not quite describing what that really returns. Also I'm not 100% sure the `|| nil` is quite right - I think it will default the value to nil if the early result type is false as well. I think I can see why you've done it to protect updating results correctly pre migration?

Maybe it would be wise to have a feature flag here so we don't change the analytics over until we switch on the new functionality, which should hopefully be quite easy to add but might affect your tests a bit!

The other thing I'd love to see is a couple more tests around the update part - in JourneyLoggerServiceSpec there are some tests at the bottom around line 163:

 ```
context "when details change" do
      let(:session_data) { { partner: false } }

      it "updates an existing record" do
        existing_record = FactoryBot.create(:completed_user_journey, assessment_id:, partner: true)
        described_class.call(assessment_id, calculation_result, check, portal_user_office_code, {})
        expect(existing_record.reload.partner).to be false
      end
    end
```

It would be great if we can add some checks here to make sure when we update an existing early result, we update the right thing successfully - and also a test when there is an early and a full result, we update the right result successfully. Does that make sense?